s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/output/ParquetTransformer.java

```diff
@@ -0,0 +1,124 @@
+        config.put(SCHEMA_REGISTRY_URL, s3SourceConfig.getString(SCHEMA_REGISTRY_URL));
+    }
+
+    @Override
+    public Iterator<byte[]> byteArrayIterator(InputStream inputStream, String topic, S3SourceConfig s3SourceConfig) throws BadDataException {
+        try {
+            List<byte[]> result = new ArrayList<>();
+            KafkaAvroSerializer avroSerializer = AvroTransformer.createAvroSerializer(s3SourceConfig);
+            for (GenericRecord genericRecord : getParquetRecords(inputStream, topic)) {
+                result.add(avroSerializer.serialize(topic, genericRecord));
+            }
+            return result.iterator();
+        } catch (NoSuchMethodException | InvocationTargetException | InstantiationException |
+                 IllegalAccessException e) {
+            throw new BadDataException(e);
+        }
+    }
+
+    /**
+     * Reads the parquet records from the input stream.
+     * @param inputStream The input stream to read.
+     * @param topic the Kafka topic the data will be placed on.
+     * @return a list of GenericRecords extracted from the stream.
+     * @throws BadDataException if the data can not be read as a Parquet structure.
+     */
+    private List<GenericRecord> getParquetRecords(final InputStream inputStream, final String topic) throws BadDataException {
+        File parquetFile = null;
+        try {
+            parquetFile = File.createTempFile(topic + "_", ".parquet");
+            try (OutputStream outputStream = Files.newOutputStream(parquetFile.toPath())) {
+                IOUtils.copy(inputStream, outputStream);
+                final InputFile inputFile = new LocalInputFile(parquetFile.toPath());
+                try (var parquetReader = AvroParquetReader.<GenericRecord>builder(inputFile).build()) {
+                    final List<GenericRecord> records = new ArrayList<>();
+                    GenericRecord record = parquetReader.read();
+                    while (record != null) {
+                        records.add(record);
+                        record = parquetReader.read();
+                    }
+                    return records;
+                }
+            } catch (RuntimeException e) { // NOPMD
+                throw new BadDataException(e);
+            }
+        } catch (IOException e) {
+            throw new BadDataException(e);
+        } finally {
+            if (parquetFile != null) {
+                deleteTmpFile(parquetFile.toPath());
+            }
+        }
+    }
+
+    /**
+     * Deletes a temporary file if it exists and logs any exceptions hit in the process.
+     * @param tempFile the file to delete.
+     */
+    // package private for testing.
+    static void deleteTmpFile(final Path tempFile) {
+        if (Files.exists(tempFile)) {
+            try {
+                Files.delete(tempFile);
+            } catch (IOException e) {
+                LOGGER.error("Error in deleting tmp file " + e.getMessage());
```

Comment:
LOGGER.error("Error in deleting tmp file {}", e.getMessage());

LOGGER.error("Error in deleting tmp file ", e.);